### PR TITLE
Show machine in dependency graph

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/59082

### CODE REVIEW

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -269,6 +269,14 @@ sub name {
     return $self->{_name};
 }
 
+sub label {
+    my ($self) = @_;
+
+    my $test    = $self->TEST;
+    my $machine = $self->MACHINE;
+    return $machine ? "$test\@$machine" : $test;
```
Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: how can we have not a `machine`?

Comment (Martchus) [reply lib/OpenQA/Schema/Result/Jobs.pm]: The code a few lines above in `sub name` implies that `MACHINE` is optional. There are also various other places, e.g. `sub latest_jobs` where `MACHINE` is optional while most other settings are mandatory. So I made this consistent with most other places in openQA. But there are indeed a *few* places where we expect machine to be present. Do you want me to fix all places to assume `MACHINE` is always present?

Comment (coolo) [reply lib/OpenQA/Schema/Result/Jobs.pm]: well, the iso controller surely won't create jobs without `MACHINE` - possibly the jobs controller allows that? Jobs created through that, won't have dependencies though :)

If you required it, you would have to pick one in the jobs controller - so having it optional might be easy enough.

Comment (Martchus) [reply lib/OpenQA/Schema/Result/Jobs.pm]: Jobs created through the jobs controller can have dependencies. The clone job script makes use of this. And after a brief look it doesn't seem to require `MACHINE` which is by the way nullable

```
    MACHINE => {
        data_type   => 'text',
        is_nullable => 1
    },
```

in contrast to other variables which are either mandatory like

```
    TEST => {
        data_type => 'text'
    },
```

or by default empty, e.g.

```
    DISTRI => {
        data_type     => 'text',
        default_value => ''
    },
```
.

**Pull request merged**